### PR TITLE
Blog Form

### PR DESCRIPTION
**Pull request by Nafisay**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -22,6 +23,7 @@ function App() {
         <ScrollToTop>
           <AuthProvider>
             <Navbar />
+            <BlogForm />
```
Comment (harshkhandeparkar) [inline src/App.tsx]: Add it to the router.

File: src/pages/BlogForm.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+function BlogForm() {
+  const authContext = useAuthContext();
+  const [error, setError] = useState<string | null>(null);
+  const [info, setInfo] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const fields: any = {
+    blog: {
+      field: "Submission Link",
+      placeholder: "",
```
Comment (harshkhandeparkar) [inline src/pages/BlogForm.tsx]: Add any placeholder, like a medium link.

File: src/pages/BlogForm.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import Form from "../components/Form";
+import { useAuthContext } from "../util/auth";
+import { makeRequest } from "../util/backend";
+
+function BlogForm() {
+  const authContext = useAuthContext();
+  const [error, setError] = useState<string | null>(null);
+  const [info, setInfo] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
```
Comment (harshkhandeparkar) [inline src/pages/BlogForm.tsx]: Add a condition here to redirect to home page if the user is not logged in.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (Nafisay) [reply src/App.tsx]: i put `STUDENT_DASHBOARD` as path

**Pull request merged**